structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+        self.width: float = width
+
+        # Calculating gross section properties
+        # Gross area
+        area: float = height * width
+        perimeter: float = 2 * (height + width)
+        ea: float = area * geometry.material.get_tangent(eps=0)[0]
+        for geo in self.geometry.geometries:
+            pass
+        for geo in self.geometry.point_geometries:
+            pass
+
+        if geometry.density is not None:
+            mass = area * geometry.density * 1e-9
+        else:
+            mass = 0
+
+
+        gp = s_res.GrossProperties()
+        gp.area = area
+        gp.perimeter = perimeter
+        gp.ea = ea
+        gp.mass = mass
+        # TODO: more gross sectional properties, and store them in self._gross_properties
+
+    @property
+    def area(self) -> float:
+        """Returns Area of the cross section.
+        Only the gross area of concrete is computed.
+
+        Returns:
+            float: The area of concrete section.
+        """
+        return self.width * self.height
+
+
+class RectangularRC(Section):
+    """Class for RC rectangular cross section.
+    The class assumes width and height respectively
+    along local x and y axes.
+
+    Input parameters:
+    width: width of the section
+    height: height of the section
+    cover: net cover (distance from outer edget of the section to
+    edge of the stirrup)
+    reinforcement_data: ReinforcementData
+    concrete: Concrete
+    reinforcement: Reinforcement (TODO!!!!!! for now const law)
+    confined_concrete: Concrete (default = None) When set to None the
+    same concrete is adopted for concrete core. When using the string
+    "auto" the confinement will be automatically computed
+    stirrups_data: StirrupsData (default = None) to use when need
+    automatic computation of concrete confinement
+    name: optional string for identidying the section
+
+    Axes definition:
+
+                     z ▲
+                       │
+                       │
+            ┌──────────┼───────────┐
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          └───────────┼─────►
+            │                      │     y
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            └──────────────────────┘
+
+            RectangularRCSection.fromGeometry(Geometry(itPolygons,itMaterials))
+
+
+    """
+
+    def __init__(
+        self,
+        width: float,
+        height: float,
+        cover: float,
+        reinforcement_data: ReinforcementData,
+        concrete: Concrete,
+        reinforcement: ElasticPlastic,
```

Comment:
I would mimic here the idea of `GenericSection`: we can input a `Material` or a `ConstitutiveLaw`. The `ConstitutiveLaw` is stored in the class.